thorvg/examples/thorvg-example/main/display_main.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include <stdlib.h>
+#include <dirent.h>
+#include "esp_log.h"
+#include "esp_err.h"
+#include "esp_spiffs.h"
+#include "bsp/esp-bsp.h"
+#include "thorvg_capi.h"
+
+static const char *TAG = "example";
+
+static void capi_shape_example(bsp_lcd_handles_t *lcd_panel);
+static void capi_lottie_example(bsp_lcd_handles_t *lcd_panel);
+static void argb888_to_rgb565(const uint32_t *argb888, uint16_t *rgb565, size_t size);
+
+#define LOTTIE_SIZE_HOR     300
+#define LOTTIE_SIZE_VER     300
+
+void app_main(void)
+{
+    bsp_lcd_handles_t lcd_panel;
+
+    bsp_spiffs_mount();
+
+    /* Initialize display */
+    bsp_display_new_with_handles(NULL, &lcd_panel);
+
+    bsp_display_backlight_on();
+
+    capi_shape_example(&lcd_panel);
+
+    vTaskDelay(pdMS_TO_TICKS(5000));
+
+    capi_lottie_example(&lcd_panel);
+}
+
+static void capi_shape_example(bsp_lcd_handles_t *lcd_panel)
+{
+    ESP_LOGI(TAG, "capiShapes start");
+
+    uint32_t *cavas_buf_888 = NULL;
+    uint16_t *cavas_buf_565 = NULL;
+
+    cavas_buf_888 = heap_caps_malloc(LOTTIE_SIZE_HOR * LOTTIE_SIZE_VER * sizeof(uint32_t), MALLOC_CAP_SPIRAM);
+    assert(cavas_buf_888);
+
+    cavas_buf_565 = heap_caps_malloc(LOTTIE_SIZE_HOR * LOTTIE_SIZE_VER * sizeof(uint16_t), MALLOC_CAP_SPIRAM);
+    assert(cavas_buf_565);
+
+    assert(tvg_engine_init(TVG_ENGINE_SW, 0) == TVG_RESULT_SUCCESS);
```

Comment:
Please don't call a function inside the `assert`. If `NDEBUG` is enabled, `tvg_engine_init` won't even get run